src/client/components/vision2/camera/distance.ts

```diff
@@ -0,0 +1,40 @@
+import { computed } from 'mobx'
+import * as THREE from 'three'
+
+import { range } from '../../../../shared/base/range'
+import { Vector4 } from '../../../math/vector4'
+import { group } from '../../three/builders'
+import { Canvas } from '../../three/three'
+import { CameraParams } from '../camera/model'
+
+import { LineProjection } from './line_projection'
+
+export class DistanceViewModel {
+  constructor(
+    private readonly params: CameraParams,
+    private readonly lineProjection: LineProjection,
+    private readonly majorStepDistance: number,
+    private readonly minorLines: number,
+    private readonly maxDistance: number,
+  ) {
+  }
+
+  static of(canvas: Canvas, params: CameraParams): DistanceViewModel {
+    return new DistanceViewModel(params, LineProjection.of(canvas, params.lens), 1.0, 3, 5)
```

Comment:
As soon as I see primitives with no labels I suggest switching to object syntax, much more readable at the call site:

```patch
diff --git a/src/client/components/vision2/camera/distance.ts b/src/client/components/vision2/camera/distance.ts
index 389396a9..b8970460 100644
--- a/src/client/components/vision2/camera/distance.ts
+++ b/src/client/components/vision2/camera/distance.ts
@@ -10,31 +10,48 @@ import { CameraParams } from '../camera/model'
 import { LineProjection } from './line_projection'
 
 export class DistanceViewModel {
-  constructor(
-    private readonly params: CameraParams,
-    private readonly lineProjection: LineProjection,
-    private readonly majorStepDistance: number,
-    private readonly minorLines: number,
-    private readonly maxDistance: number,
-  ) {
+  private readonly params: CameraParams
+  private readonly lineProjection: LineProjection
+  private readonly majorStepDistance: number
+  private readonly minorLines: number
+  private readonly maxDistance: number
+
+  constructor({ params, lineProjection, majorStepDistance, minorLines, maxDistance }: {
+    params: CameraParams,
+    lineProjection: LineProjection,
+    majorStepDistance: number,
+    minorLines: number,
+    maxDistance: number,
+  }) {
+    this.params = params
+    this.lineProjection = lineProjection
+    this.majorStepDistance = majorStepDistance
+    this.minorLines = minorLines
+    this.maxDistance = maxDistance
   }
 
   static of(canvas: Canvas, params: CameraParams): DistanceViewModel {
-    return new DistanceViewModel(params, LineProjection.of(canvas, params.lens), 1.0, 3, 5)
+    return new DistanceViewModel({
+      params,
+      lineProjection: LineProjection.of(canvas, params.lens),
+      majorStepDistance: 1,
+      minorLines: 3,
+      maxDistance: 5,
+    })
   }
 
   readonly distance = group(() => ({
     children: range((this.minorLines + 1) * this.maxDistance / this.majorStepDistance).map(i => (
-        this.lineProjection.cone({
-          axis: this.params.Hcw.z.vec3().multiplyScalar(-1),
-          radius: Math.cos(Math.atan(((i + 1) / (this.minorLines + 1)) / this.height)),
-          color: new Vector4(1, 1, 1, (i + 1) % (this.minorLines + 1) ? 0.2 : 0.4),
-          lineWidth: (i + 1) % (this.minorLines + 1) ? 2 : 3,
-        }))),
+      this.lineProjection.cone({
+        axis: this.params.Hcw.z.vec3().multiplyScalar(-1),
+        radius: Math.cos(Math.atan(((i + 1) / (this.minorLines + 1)) / this.height)),
+        color: new Vector4(1, 1, 1, (i + 1) % (this.minorLines + 1) ? 0.2 : 0.4),
+        lineWidth: (i + 1) % (this.minorLines + 1) ? 2 : 3,
+      }))),
   }))
 
   @computed
-  private get height() : number {
+  private get height(): number {
     return new THREE.Matrix4().getInverse(this.params.Hcw.toThree()).elements[15]
   }
 }

```

looks like auto-format also did some things